build.zig

```diff
@@ -782,7 +782,13 @@ fn addCmakeCfgOptionsToExe(
                 mod.linkSystemLibrary("unwind", .{});
             },
             .ios, .macos, .watchos, .tvos, .visionos => {
-                mod.link_libcpp = true;
+                if (static or !std.zig.system.darwin.isSdkInstalled(b.allocator)) {
+                    mod.link_libcpp = true;
+                } else {
+                    const sdk = std.zig.system.darwin.getSdk(b.allocator, b.graph.host.result) orelse return error.SdkDetectFailed;
+                    const @"libc++" = b.pathJoin(&.{ sdk, "usr/lib/libc++.tbd" });
+                    exe.root_module.addObjectFile(.{ .cwd_relative = @"libc++" });
+                }
```

Comment:
Instead of using `isSdkInstalled()`, can't we just call `getSdk()` and check the result for null? That'd be 1 child process invocation instead of 2.

Reply:
The 2 existing uses (call sites) of `getSdk()` are preceded by `isSdkInstalled()` to avoid a desktop popup window (iirc) for agreeing to xcode license. I think it's needed for the build executable as well.

I would be nice if we figured out a way to put this kind of thing in zig-cache but the difficulty is in what is the "dependency" and how to detect when the SDK is changed.

Reply:
Ok, that kinda sucks, but makes sense.

Reply:
... a half-baked thought I had on that is what if we simply used a cookie (hash) to created the idea of a session, where that is the root zig process. The hash could be against { zig-version, PID, and timestamp }. But it would need to be communicated to child processes.